recipes/yapss/meta.yaml

```
@@ -0,0 +1,144 @@
+{% set name = "yapss" %}
+{% set version = "0.1.0" %}
+{% set sha256 = "ef084862f31dddd2c16e53ff60014768a64fb9e7266d75af5477dd3f1f44e405" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/y/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  noarch: python
+  number: 0
+
+requirements:
+  host:
+    - python >=3.9,<3.14
+    - hatchling
+    - hatch-vcs
+    - pip
+  run:
+    - python >=3.9,<3.14
+    - numpy
+    - casadi >=3.6.0
+    - scipy
+    - mpmath
+    - matplotlib-base
+    - cyipopt
+
+outputs:
+  - name: yapss
+    build:
+      script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+    requirements:
+      host:
+        - python >=3.9,<3.14
+        - hatchling
+        - hatch-vcs
+        - pip
+      run:
+        - python >=3.9,<3.14
+        - numpy
+        - casadi >=3.6.0
+        - scipy
+        - mpmath
+        - matplotlib-base
+        - cyipopt
+    test:
+      source_files:
+        - tests/examples/test_hs071.py
+      requires:
+        - pytest
+      commands:
+        - pytest tests/
+    about:
+      home: https://github.com/stevenrhall/yapss
+      summary: "A Python package for solving optimal control problems using pseudospectral methods."
+      description: |
+        YAPSS is a Python package for numerically solving multi-phase optimal control problems using
+        pseudospectral methods, similar to the GPOPS-II algorithm. The package supports multiple
+        differentiation methods, including automatic differentiation using CasADi, as well as numerical
+        derivatives found by central difference methods for problems that cannot be differentiated by
+        automatic methods. YAPSS supports multiple collocation methods, including Legendre-Gauss (LG),
+        Legendre-Gauss-Radau (LGR), and Legendre-Gauss-Lobatto (LGL) collocation points. The mesh
+        structure can be specified by the user to improve the accuracy of the solution.
+      license: MIT
+      license_file: LICENSE
+      doc_url: https://yapss.readthedocs.io
+      dev_url: https://github.com/stevenrhall/yapss
+
+  - name: yapss-dev
+    build:
+      script: "{{ PYTHON }} -m pip install .[dev] --no-deps -vv"
+    requirements:
+      host:
+        - python >=3.9,<3.14
+        - hatchling
+        - hatch-vcs
+        - pip
+      run:
+        # yapss dependencies
+        - python >=3.9,<3.14
+        - numpy
+        - casadi >=3.6.0
+        - scipy
+        - mpmath
+        - matplotlib-base
+        - cyipopt
+        # documentation
+        - sphinx
+        - sphinxcontrib-bibtex
+        - sphinx-copybutton
+        - sphinx_rtd_theme
+        - myst-parser
+        - nbsphinx
+        - numpydoc
+        - furo
+        # notebooks
+        - jupyterlab
+        - jupyterlab_code_formatter
+        - jupyterlab-spellchecker
+        - ipython
+        - black[jupyter] >=24.0.0,<25.0.0
```

Brackets have no effect in conda 
```suggestion
        - black >=24.0.0,<25.0.0
```

---

Done. I added the [jupyter] dependencies separately.